src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionCloud.java

```diff
@@ -289,9 +293,68 @@ public Credential createAccessToken(URL broker) throws AbortException, OauthClie
         }
         label = Jenkins.getInstance().getLabel(st.getLabel()+" "+box.size+compat);
 
-        
+
         final Queue<PlannedMansionSlave> queue = new ArrayBlockingQueue<PlannedMansionSlave>(excessWorkload);
         List<PlannedNode> r = new ArrayList<PlannedNode>();
+
+        /**
+         * The current approach - fire up as much slaves as we can - is pretty broken.
```

Comment:
If you want a comment here I would just say why you are throttling not why things were the way they were before... that is a discussion for the commit message if you feel strongly about documenting such